rust/src/error.rs

```diff
@@ -0,0 +1,193 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! ADBC error enums and structs
+
+use arrow_schema::ArrowError;
+
+#[derive(Debug, PartialEq, Copy, Clone)]
+#[repr(u8)]
+pub enum AdbcStatusCode {
+    /// No error.
+    Ok = 0,
```

Comment:
Does this make sense to have in Rust? It seems `Ok` would be expressed as `Result::Ok` right?

Reply:
This part is meant to be FFI compatible with the C error codes (probably should have documented that). Although yeah this does provide a redundant state.

Maybe the best solution is to drop this value from the Rust enum, and then handle the conversion of `0` into `Result::Ok` within the C API FFI codebase. 🤔

Reply:
Yes, keep the Rust definitions appropriate for Rust and do any conversions needed in the FFI part. (Again: the Go and Java definitions make no reference to FFI, though they are still structured as to make FFI reasonable. But Rust should be able to use Rust only without ever having to consider FFI.)